plugins/s3_auth/aws_auth_v4.cc

```diff
@@ -99,6 +101,33 @@ uriEncode(const String in, bool isObjectName)
 }
 
 /**
+ * @brief URI-decode a character string (AWS specific version, see spec)
+ *
+ * @see AWS spec: http://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html
+ *
+ * @todo Consider reusing / converting to TSStringPercentDecode()
```

Comment:
Maybe add to the comment as to why we can't use TSStringPercetDecode() ? I assume we can't? :-)

Reply:
AWS URI-encoding is not standard but AFAICT functionally it seems possible to reuse `TSStringPercent(En|De)code()` (even if we need to make changes).

It seems we are not building a library that will allow the unit-test binary to use `TSStringPercent(En|De)code()` and to make the necessary changes to create one would require more time. Also it did not seem a good idea to use one set of encoding/decoding functions during runtime and another during unit-tests.

I will add a note in the comments.